core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/codeGen/ReplCodeGeneratorImpl.kt

```diff
@@ -87,13 +86,15 @@ internal class ReplCodeGeneratorImpl : ReplCodeGenerator {
         isOpen: Boolean,
     ): CodeWithConverter {
         val result = generator.generate(
-            schema,
-            name,
+            schema = schema,
+            name = name,
             fields = false,
             extensionProperties = true,
-            isOpen,
-            MarkerVisibility.IMPLICIT_PUBLIC,
-            registeredMarkers.values
+            isOpen = isOpen,
+            visibility = MarkerVisibility.IMPLICIT_PUBLIC,
+            knownMarkers = registeredMarkers
+                .filterKeys { !it.isData } // filter out data classes, so they aren't used as supertypes
```

Comment:
this also should be filtered out
```kotlin
@DataSchema(isOpen = false)
class A(val a: Int)
```

Reply:
oh right! Better check if it's open

Reply:
Even then, an interface cannot inherit from a class, so
```@DataSchema(isOpen = true)
open class A(val a: Int)
```
should not be inherited from (using an interface)

Reply:
plus there's the constructor etc...